main.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/ONSdigital/rm-survey-service/models"
+	"github.com/cloudfoundry-community/go-cfenv"
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+)
+
+const (
+	realm       = "sdc"
+	serviceName = "surveysvc"
+	timeFormat  = "2006-01-02T15:04:05Z0700"
+)
+
+var logger *zap.Logger
+
+func init() {
+	logger, _ = zap.NewProduction()
+	defer logger.Sync()
+}
+
+func main() {
+	dataSource, port := configureEnvironment()
+	db := models.InitDB(dataSource)
+
+	// Set up the signal handler to watch for SIGTERM and SIGINT signals so we
```

Comment:
we need to handle the OS signals mentioned here - use the os/signals package for this

Reply:
See the example https://github.com/ONSdigital/git-diff-check-service/blob/alpha/main.go#L37 - obviously fill in appropriate shut down routines for your bits like db connections and queries in progress (especially any transactions - may want to look at go contexts for that)

Reply:
have raised a github issue for this and has been added to our backlog